.github/workflows/check_bom.yml

```diff
@@ -0,0 +1,43 @@
+name: BOM Check
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  bom-check:
+    if: ${{ github.event_name != 'pull_request' || github.repository != github.event.pull_request.head.repo.full_name }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: UTF-8 BOM
+        run: | 
+          grep -oarlE -m 1 --include="*.{c,h,rst,txt,mk,py,cmake}" --include="Makefile" $'\xEF\xBB\xBF' . | while read -r line; do
+            a=$(head -c 3 "$line")
+            if [ "${a}" = $'\xEF\xBB\xBF' ]; then
+              echo "${line}" && exit 1
+            fi
+          done
+        
+      - name: UTF-16LE BOM
+        run: |
+          grep -oarlE -m 1 --include="*.{c,h,rst,txt,mk,py,cmake}" --include="Makefile" $'\xFF\xFE' . | while read -r line; do
+            a=$(head -c 3 "$line")
+            if [ "${a}" = $'\xFF\xFE' ]; then
+              echo "${line}" && exit 1
+            fi
+          done
+
+      - name: UTF-16BE BOM
+        run: |
+          grep -oarlE -m 1 --include="*.{c,h,rst,txt,mk,py,cmake}" --include="Makefile" $'\xFE\xFF' . | while read -r line; do
```

Comment:
I tested it in my own branch and unfortunately it did not catch a BOM until I changed the `--include`. Seems like the GLOB using `{}` isn't valid.

```suggestion
        run: |
          grep -oarlE -m 1 --include="*.c" --include="*.h" --include="*.rst" --include="*.txt" --include="*.mk" --include="*.py" --include="*.cmake" --include="Makefile" $'\xEF\xBB\xBF' . | while read -r line; do
            a=$(head -c 3 "$line")
            if [ "${a}" = $'\xEF\xBB\xBF' ]; then
              echo "${line}" && exit 1
            fi
          done

      - name: UTF-16LE BOM
        run: |
          grep -oarlE -m 1 --include="*.c" --include="*.h" --include="*.rst" --include="*.txt" --include="*.mk" --include="*.py" --include="*.cmake" --include="Makefile" $'\xFF\xFE' . | while read -r line; do
            a=$(head -c 3 "$line")
            if [ "${a}" = $'\xFF\xFE' ]; then
              echo "${line}" && exit 1
            fi
          done

      - name: UTF-16BE BOM
        run: |
          grep -oarlE -m 1 --include="*.c" --include="*.h" --include="*.rst" --include="*.txt" --include="*.mk" --include="*.py" --include="*.cmake" --include="Makefile" $'\xFE\xFF' . | while read -r line; do
```

Reply:
I was just doing as I was instructed. I know the previous code did work so I will revert the commit for this change.

Reply:
I've asked ChatGPT to simplify it. I should have tested it. Sorry. @kdschlosser please use Liam's suggestion as it's tested and simpler.